src/mail-app/workerUtils/offline/MailOfflineCleaner.ts

```diff
@@ -55,20 +56,46 @@ export class MailOfflineCleaner implements OfflineStorageCleaner {
 						await this.deleteMailSetEntries(offlineStorage, mailSet.entries, customCutoffId)
 					}
 				}
-				// TODO MailSet cleanup
-				const mailListIds = [mailBox.currentMailBag!, ...mailBox.archivedMailBags].map((mailbag) => mailbag.mails)
-				for (const mailListId of mailListIds) {
-					await this.deleteMailListLegacy(offlineStorage, mailListId, cutoffId)
-				}
 			}
 		}
 	}
 
 	/**
-	 * This method deletes mails from {@param listId} what are older than {@param cutoffId} as well as associated data.
-	 *
-	 * it's considered legacy because once we start importing mail into mail bags, maintaining mail list ranges doesn't make
-	 * sense anymore - mail order in a list is arbitrary at that point.
+	 * delete all mail set entries of a mail set that reference some mail with a receivedDate older than
```

Comment:
The comment is outdated now, but it underlines important point, we should make sure we do not cache mails based on list id. The only place I can think that does this is `BulkMailLoader`. We should check that it does not use caching entity client for loading mails in ranges.